tabpy-server/tabpy_server/app/app.py

```diff
@@ -0,0 +1,310 @@
+
+        # set and validate transfer protocol
+        set_parameter('transfer_protocol',
+                      ConfigParameters.TABPY_TRANSFER_PROTOCOL,
+                      default_val='http')
+        self.settings['transfer_protocol'] =\
+            self.settings['transfer_protocol'].lower()
+
+        set_parameter('certificate_file',
+                      ConfigParameters.TABPY_CERTIFICATE_FILE)
+        set_parameter('key_file', ConfigParameters.TABPY_KEY_FILE)
+        self._validate_transfer_protocol_settings()
+
+        # if state.ini does not exist try and create it - remove
+        # last dependence on batch/shell script
+        set_parameter('state_file_path', ConfigParameters.TABPY_STATE_PATH,
+                      default_val='./', check_env_var=True)
+        self.settings['state_file_path'] = os.path.realpath(
+            os.path.normpath(
+                os.path.expanduser(self.settings['state_file_path'])))
+        state_file_path = self.settings['state_file_path']
+        logger.info("Loading state from state file %s" %
+                    os.path.join(state_file_path, "state.ini"))
+        tabpy_state = _get_state_from_file(state_file_path)
+        self.tabpy_state = TabPyState(
+            config=tabpy_state, settings=self.settings)
+
+        self.python_service = PythonServiceHandler(PythonService())
+        self.settings['compress_response'] = True if TORNADO_MAJOR >= 4\
+            else "gzip"
+        self.settings['static_path'] = os.path.join(
+            os.path.dirname(__file__), "static")
+
+        # Set subdirectory from config if applicable
+        if tabpy_state.has_option("Service Info", "Subdirectory"):
+            self.subdirectory = "/" + \
+                tabpy_state.get("Service Info", "Subdirectory")
+
+        # If passwords file specified load credentials
+        set_parameter(ConfigParameters.TABPY_PWD_FILE,
+                      ConfigParameters.TABPY_PWD_FILE)
+        if ConfigParameters.TABPY_PWD_FILE in self.settings:
+            if not self._parse_pwd_file():
+                log_and_raise('Failed to read passwords file %s' %
+                              self.settings[ConfigParameters.TABPY_PWD_FILE],
+                              RuntimeError)
+        else:
+            logger.info("Password file is not specified")
+
+        features = self._get_features()
+        self.settings['versions'] = {'v1': {'features': features}}
+
+    def _validate_transfer_protocol_settings(self):
+        if 'transfer_protocol' not in self.settings:
+            log_and_raise(
+                'Missing transfer protocol information.', RuntimeError)
+
+        protocol = self.settings['transfer_protocol']
+
+        if protocol == 'http':
+            return
+
+        if protocol != 'https':
+            log_and_raise('Unsupported transfer protocol: {}.'.format(
+                protocol), RuntimeError)
+
+        self._validate_cert_key_state('The parameter(s) {} must be set.',
```

Comment:
The {} here and line 267 don't appear to be needed since this not being formatted

Reply:
`_validate_cert_key_state` which is called here does the formatting